addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5250,51 +5279,99 @@ registry.layout_column = SnippetOptionWidget.extend({
      * last x columns.
      *
      * @private
-     * @param {jQuery} $row - the row in which to update the columns
-     * @param {integer} count - positif to add, negative to remove
+     * @param {HTMLElement} rowEl - the row in which to update the columns
+     * @param {integer} nbColumns - the number of columns requested
      */
-    _updateColumnCount: async function ($row, count) {
-        if (!count) {
+    async _updateColumnCount(rowEl, nbColumns) {
+        // itemsDelta: positive to add, negative to remove
+        const itemsDelta = nbColumns - rowEl.children.length;
+        const isMobile = weUtils.isMobile(this);
+        const mobileColumns = this._getNbColumns(true);
```

Comment:
Before digging deeper in this implementation, I'd like to ask @pierre-pde if it is really what he wants functionally,  there are many thing which are weird to me:
- Set up 3 column in desktop, it shows "3", go to mobile, make two of them side by side: it show 2 if you did it with the first 2 but 1 if you did it with the last 2.
- Why is the logic different between desktop and mobile:
    - Desktop: choose 3 columns, resize one so it wraps over 2 lines -> it still shows 3
    - Mobile: choose 3 columns, resize one so it wraps over 2 lines -> it shows 2 (or 1 as explained above)
- Bug: desktop choose 3 columns, resize the right one to make it bigger: does not work immediately (no issue in standard master)
- IMO, the order you set up on desktop/mobile should stay (kinda like what we do with grid). If I set up 3 columns in desktop: image - text -image, then in mobile set up the 2 images to be side-by-side and the text below... it should not impact the desktop order IMO. The logic may not be very easy but there is a way to do it thanks to order-XXX classes.

I think there is more to test and even more to do here :wink: